diff --git a/pyproject.toml b/pyproject.toml
index 33c759e..c6ead00 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -17,8 +17,6 @@ classifiers = [
   "Programming Language :: Python :: 3.11",
   "Programming Language :: Python :: 3.12",
   "Programming Language :: Python :: 3.13",
-  "Programming Language :: Python :: 3.14",
-  "Programming Language :: Python :: 3.15",
   "Operating System :: OS Independent",
   "License :: OSI Approved :: GNU Lesser General Public License v3 (LGPLv3)"
 ]
