From 0a636732c745bd36138103b941f1c62882149b52 Mon Sep 17 00:00:00 2001
From: Rafael Beims <rafael.beims@toradex.com>
Date: Mon, 29 May 2023 18:39:55 +0000
Subject: [PATCH] add install target

---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index 6c2d6e3..d645966 100755
--- a/Makefile
+++ b/Makefile
@@ -577,6 +577,9 @@ else
 default:
 	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) modules
 
+modules_install:
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) INSTALL_MOD_PATH=$(MODLIB) modules_install
+
 endif
 
 ###############################################################
-- 
2.30.2

