From 98b446cef2eb6b813bbb8622104d6f868644dcbe Mon Sep 17 00:00:00 2001
From: Rafael Beims <rafael.beims@toradex.com>
Date: Wed, 21 Jun 2023 17:47:12 +0000
Subject: [PATCH] Remove strip from the build

---
 DutApiWiFiBt/MakeFile_W8997_FC18 | 1 -
 1 file changed, 1 deletion(-)

diff --git a/DutApiWiFiBt/MakeFile_W8997_FC18 b/DutApiWiFiBt/MakeFile_W8997_FC18
index 7639d5e..61bfb03 100644
--- a/DutApiWiFiBt/MakeFile_W8997_FC18
+++ b/DutApiWiFiBt/MakeFile_W8997_FC18
@@ -95,7 +95,6 @@ endif
 
 labtool:	$(LABTOOL_OBJS)		
 		$(LINK) $(CFLAGS) $(LABTOOL_OBJS) -lpthread $(DRVR_STACK) -lm
-		$(STRIP) $@
 
 		#cp labtool /tftpboot/bulverde/root/
 
-- 
2.30.2

